rex/rechunk_h5/chunk_size.py

```diff
@@ -0,0 +1,259 @@
+"""
+Module to rechunk existing .h5 files
+"""
+from abc import ABC
+import numpy as np
+
+
+class BaseChunkSize(ABC):
+    """
+    Base class to compute chunk size
+    """
+    def __init__(self):
+        """
+        Parameters
+        ----------
+        chunk_size : int, optional
+            Chunk size in MB, by default 2
+        """
+        self._chunks = None
+
+    @property
+    def chunks(self):
+        """
+        Dataset chunk size along all axis
+
+        Returns
+        -------
+        tuple
+        """
+        return self._chunks
+
+
+class TimeseriesChunkSize(BaseChunkSize):
+    """
+    Compute Timeseries chunks based on dtype, and weeks_per_chunk
+    """
+    def __init__(self, shape, dtype, chunk_size=2, weeks_per_chunk=None):
+        """
+        Parameters
+        ----------
+        shape : tuple
+            Array shape
+        dtype : str | np.dtype
+            Array data type
+        chunk_size : int, optional
+            Chunk size in MB, by default 2
+        weeks_per_chunk : int, optional
+            Number of weeks per time chunk, if None scale weeks based on 8
+            weeks for hourly data, by default None
+        """
+        self._chunks = self.compute_dtype_chunks(
+            shape, dtype,
+            chunk_size=chunk_size,
+            weeks_per_chunk=weeks_per_chunk)
+
+    @staticmethod
+    def _compute_time_chunk_size(t_len, weeks_per_chunk=8):
+        """
+        Compute the size of the temporal chunks. Chunks are based on
+
+        Parameters
+        ----------
+        t_len : int
+            Length of temporal axis
+        weeks_per_chunk : int, optional
+            Number of weeks per time chunk, if None scale weeks based on 8
+            weeks for hourly data, by default None
+
+        Returns
+        -------
+        time_chunks : int
+            Size of chunks along temporal axis
+        """
+        # Freq is the number of time-steps per hour
+        freq = t_len / 8760
+
+        if weeks_per_chunk is None:
+            # Use default of 8 weeks for hourly chunks
+            weeks_per_chunk = np.ceil(8 / freq)
+
+        # Compute time chunks
+        time_chunk = weeks_per_chunk * 7 * 24 * freq
+
+        return int(time_chunk)
+
+    @staticmethod
+    def _compute_site_chunk_size(time_chunk, dtype, chunk_size=2):
+        """
+        Compute spatial chunk size given time chunk size
+
+        Parameters
+        ----------
+        time_chunk : int
+            Lenght of chunk along the temporal axis
+        dtype : str | np.dtype
+            Array data type (dtype)
+        chunk_size : int, optional
+            Chunk size in MB, by default 2
+
+        Returns
+        -------
+        site_chunk : int
+            Spatial chunk size or size of chunk along non-temporal axes
+        """
+        pixel_size = np.dtype(dtype).itemsize * 10**-6
+        site_chunk = chunk_size // (time_chunk * pixel_size)
+
+        return int(site_chunk)
```

Comment:
Wont this round to zero if time chunk is very large? Like if weeks per chunk is big or if its high temporal res data? I think we should always use ceil() for these methods instead of int()

Reply:
Good call. I also need to add a check such that time_chunk > len(arr)